src/miniscript/context.rs

```diff
@@ -353,6 +394,12 @@ impl ScriptContext for Segwitv0 {
                 }
                 Ok(())
             }
+            Terminal::Multi(_k, ref pks) => {
```

Comment:
This seems like a new rule for segwit v0... separate PR?

(side note: I'm a bit confused as to why these functions are all parametrized by Ctx vs. Self?)

Reply:
It's fine for it to be in this PR.

Reply:
Indeed this is a separate bug fix, I will make it under a separate commit. This prevents the usage of uncompressed keys in segwitv0 context.

> I'm a bit confused as to why these functions are all parametrized by Ctx vs. Self

This is because we never have an instance of Self in our usage of ScriptContext, but is always used as a generic in Miniscript type itself. For example, if the user writes `let ms = Miniscript::<String, Tap>::from_str(multi(...))` we can sanely error. And we prevent using miniscripts across contexts. I don't see how we can do that if we have a self parameter.

Reply:
it's off topic for this pr, but:

not a self parameter a Self type. As in

```
pub trait ScriptContext:
    fmt::Debug + Clone + Ord + PartialOrd + Eq + PartialEq + hash::Hash + private::Sealed
{
    fn check_terminal_non_malleable<Pk: MiniscriptKey, Ctx: ScriptContext>(
        _frag: &Terminal<Pk, Ctx>,
    ) -> Result<(), ScriptContextError>;
```

should be

```rust
pub trait ScriptContext:
    fmt::Debug + Clone + Ord + PartialOrd + Eq + PartialEq + hash::Hash + private::Sealed
{
    fn check_terminal_non_malleable<Pk: MiniscriptKey>(
        _frag: &Terminal<Pk, Self>,
    ) -> Result<(), ScriptContextError>;
```

It seems like otherwise the type of ScriptContext permittable into  any of the functions is *any* ScriptContext, not the specific one we just defined it for?

Reply:
Indeed, this is an improvement.

Reply:
Ok I can do this as a follow up commit after this goes through -- hopefully not too annoying.